feature-libs/product-configurator/rulebased/components/attribute/types/input-field/configurator-attribute-input-field.component.ts

```diff
@@ -19,15 +23,40 @@ import { ConfiguratorAttributeBaseComponent } from '../base/configurator-attribu
 })
 export class ConfiguratorAttributeInputFieldComponent
   extends ConfiguratorAttributeBaseComponent
```

Comment:
This `ConfiguratorAttributeBaseComponent` is extended in many places. Can we make this BaseComponent as a service (which is singleton)? So, instead of extend it, but inject it into required components.

Reply:
picked

Reply:
No, let's stick to the current approach. We would introduce many breaking changes to existing components